APE18.rst

```diff
@@ -0,0 +1,55 @@
+Adopt NEP 29 for CPython and Numpy Version Support
+--------------------------------------------------
+
+author: Stuart Mumford and Thomas Robitaille
+
+date-created: 2020 October 16
+
+date-last-revised: 2020 October 16
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+APE 10 previously determined supported CPython versions, and provided a list up to the 4.0 release.
+Since APE 10 was written the wider scientific Python community have written and largely adopted `NEP 29`_ as a policy for supported versions of both CPython and Numpy.
+This APE proposes that Astropy adopt this policy following other projects in the scientific ecosystem.
+
+
+Detailed description
+--------------------
+
+This APE fully adopts `NEP 29`_ for both CPython and Numpy versions, it supersedes APE 10 wherever there is a conflict between that APE and this APE or `NEP 29`_.
+
+While this APE will not replicate the whole of `NEP 29`_, and `NEP 29`_ should be considered the canonical source for the support policy, the core scheme to be adopted is as follows:
+
+The core astropy package will support:
+
+* All minor versions of Python released 42 months prior to the project, and at minimum the two latest minor versions.
+* All minor versions of numpy released in the 24 months prior to the project, and at minimum the last three minor versions.
+
```

Comment:
```suggestion

Note that, as stated in `NEP 29`_, these timelines all apply for the versions
set at the time of the non-bugfix release.  This means that the full
support timelines for LTS versions of astropy are 66 and 48 months for Python
and numpy, respectively.

```

Reply:
Maybe worth saying since these are currently supported for two years (in case we change that some day)

Reply:
I find this phrasing very confusing. 

What about rewording to say the 42 and 24 are for the start of a release "family", and support won't be dropped in a bugfix release, neither on LTS nor on normal feature release bugfix?

Reply:
(also, I promised Stu a blurb about the exceptation, e.g. the latest LTS dropped a numpy version the release, as it would have been dropped  based on NEP29 very soon after the release)

Reply:
> What about rewording to say the 42 and 24 are for the start of a release "family", and support won't be dropped in a bugfix release, neither on LTS nor on normal feature release bugfix?

:+1: This is in NEP 29, but I am happy to call it out here in terms of the LTS if people feel it needs more clarity.

Reply:
yes, I'm OK just the reference back to NEP29, it's very clear in there, but the above blurb about the 66 and 48 is very very confusing to say the least. A start of a release is a very clear date, but the end of it? And why would it even matter, by definition things are not going to be dropped between 4.0.10 and 4.0.11.

Reply:
Agreed that this adds more confusion than it solves. I'd leave the text as is.

Reply:
I disagree that it's very clear in NEP 29 for the *casual* reader.  That is it's very specific and when I read very carefully I understand the above, and really anyone in this conversation is clearly interested enough to dig deeply on this.  But I do *not* think the typical reader who comes upon this via google and just wants to know "how long will an astropy support this Python version" will take the time to click through to NEP 29, map it onto our Astropy-specific release candence, etc.

That said, if y'all think this wording is not clearer, maybe the better thing is to just give a few concrete examples and let the reader extrapolate?  Something like:
```suggestion
Note that, as stated in `NEP 29`_, these timelines all apply for the versions
set at the time of the non-bugfix release.  So for example the v4.0 release, an LTS, would support Python 3.6 (released 23 mo earlier), and would continue to support it until the end of the LTS window (24 mo after the 4.0 release).  Whereas 4.2, planned for late 2020, would  unambiguously only support 3.7 (as its full lifetime falls within the 3.7 window).
```
Or @cadirs wording from https://github.com/astropy/astropy-APEs/pull/63/files#r506678119 makes sense to me too.

Reply:
See my extra sentence about the bugfix lifetimes.

Reply:
(I don't find these concrete examples any useful, try to swap it out with e.g 1.3 and it will all read obsolete for the casual reader who you say would not bother to read/try to understand the tables and figures in NEP29)

Reply:
There is a line now stating that the versions are supported for the lifetime of the bugfixes, so hopefully that resolves the core of your comment @eteq in a way people are happy with?

Reply:
Reads well to me.